### ISSUE TITLE
ServoGraphiteSetup setup should be per type, not once for all things

### ISSUE THREAD

**Issue by jewzaam** [lightblue-platform/lightblue-core#350]
Since the types are now an argument to initialize, could handle initialization multiple times.  Open for discussion so tagging as open question.  Should we allow different types to be initialized separately?  If not why have the types as an argument if it's contingent on variables defined as properties or env vars?

**Comment by jewzaam** [comment]
Note this is the PR introducing this change: https://github.com/lightblue-platform/lightblue-core/pull/348

**Comment by luan-cestari** [comment]
@jewzaam You mentioned that the ServoGraphiteSetup will be initialized multiple times , if this is true, we should create some tests to guarantee it will work properly (like, one of the resets called inside the initialization can clean some metrics that we would not expect to happen). The unit test class https://github.com/lightblue-platform/lightblue-core/blob/6cd63679e857febf8841e5bfaaaa8699b59510fc/hystrix/src/test/java/com/redhat/lightblue/hystrix/ServoGraphiteSetupTest.java

**Comment by jewzaam** [comment]
It doesn't support multiple initialization right now.  If we decided to change it to support that, yes tests need to be added.

**Comment by luan-cestari** [comment]
Yeah. It is a singleton but it doesn't initialize with the class. As it can be loaded concurrently, I think we should put a lock or make the method synchronized or other concurrent options on https://github.com/lightblue-platform/lightblue-core/blob/6cd63679e857febf8841e5bfaaaa8699b59510fc/hystrix/src/main/java/com/redhat/lightblue/hystrix/ServoGraphiteSetup.java (or move the intialization to a static block as just a single class will be initialized by the classloader ) (I know this is another issue, we can even open it if you think it will eb better)

**Comment by dcrissman** [comment]
My primary driver for adding Type is because before if the environment variables were not there, then an INFO log would be dropped saying "GraphiteMetricObserver not configured, missing environment variable". This log would make me think that something was not configured correctly, even though my intention is not to use GraphiteMetricObserver. No need to have a nagging log.

That said, if I did want to use GraphiteMetricObserver but did not have the environment variables set, then a log would be appreciated.

Maybe instead (or in addition) we include another env var stating what kind of MetricObserver you want to use?

**Comment by bserdar** [comment]
No longer relevant.
